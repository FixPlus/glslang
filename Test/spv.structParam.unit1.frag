#version 460 core

struct MyStruct {
    vec3 v;
    int i;
};

int foo(inout MyStruct s)
{
    s.v = vec3(0.f, 1.f, 2.f);
    return s.i;
}
