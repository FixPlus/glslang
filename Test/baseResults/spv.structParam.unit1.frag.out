spv.structParam.unit1.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:8  Function Definition: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:8    Function Parameters: 
0:8      's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:10    Sequence
0:10      move second child to first child ( temp highp 3-component vector of float)
0:10        v: direct index for structure ( global highp 3-component vector of float)
0:10          's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:10          Constant:
0:10            0 (const int)
0:10        Constant:
0:10          0.000000
0:10          1.000000
0:10          2.000000
0:11      Branch: Return with expression
0:11        i: direct index for structure ( global highp int)
0:11          's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:11          Constant:
0:11            1 (const int)
0:?   Linker Objects
0:?     Linker Function: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( inout structure{ global highp 3-component vector of float v,  global highp int i})

spv.structParam.unit2.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:12  Function Definition: main( ( global void)
0:12    Function Parameters: 
0:?     Sequence
0:15      move second child to first child ( temp highp int)
0:15        i: direct index for structure ( global highp int)
0:15          's' ( temp structure{ global highp 3-component vector of float v,  global highp int i})
0:15          Constant:
0:15            1 (const int)
0:15        Constant:
0:15          2 (const int)
0:16      Sequence
0:16        move second child to first child ( temp highp int)
0:16          'r' ( temp highp int)
0:16          Function Call: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:16            's' ( temp structure{ global highp 3-component vector of float v,  global highp int i})
0:?   Linker Objects
0:?     Linker Function: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:?     'cout' ( out highp float)


Linked fragment stage:


Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:8  Function Definition: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:8    Function Parameters: 
0:8      's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:10    Sequence
0:10      move second child to first child ( temp highp 3-component vector of float)
0:10        v: direct index for structure ( global highp 3-component vector of float)
0:10          's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:10          Constant:
0:10            0 (const int)
0:10        Constant:
0:10          0.000000
0:10          1.000000
0:10          2.000000
0:11      Branch: Return with expression
0:11        i: direct index for structure ( global highp int)
0:11          's' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:11          Constant:
0:11            1 (const int)
0:12  Function Definition: main( ( global void)
0:12    Function Parameters: 
0:?     Sequence
0:15      move second child to first child ( temp highp int)
0:15        i: direct index for structure ( global highp int)
0:15          's' ( temp structure{ global highp 3-component vector of float v,  global highp int i})
0:15          Constant:
0:15            1 (const int)
0:15        Constant:
0:15          2 (const int)
0:16      Sequence
0:16        move second child to first child ( temp highp int)
0:16          'r' ( temp highp int)
0:16          Function Call: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:16            's' ( temp structure{ global highp 3-component vector of float v,  global highp int i})
0:?   Linker Objects
0:?     'cout' ( out highp float)

// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 48

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 47
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 460
                              Name 4  "main"
                              Name 9  "MyStruct"
                              MemberName 9(MyStruct) 0  "v"
                              MemberName 9(MyStruct) 1  "i"
                              Name 13  "foo(struct-MyStruct-vf3-i11;"
                              Name 12  "s"
                              Name 28  "MyStruct"
                              MemberName 28(MyStruct) 0  "v"
                              MemberName 28(MyStruct) 1  "i"
                              Name 30  "s"
                              Name 33  "r"
                              Name 34  "param"
                              Name 47  "cout"
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 3
               8:             TypeInt 32 1
     9(MyStruct):             TypeStruct 7(fvec3) 8(int)
              10:             TypePointer Function 9(MyStruct)
              11:             TypeFunction 8(int) 10(ptr)
              15:      8(int) Constant 0
              16:    6(float) Constant 0
              17:    6(float) Constant 1065353216
              18:    6(float) Constant 1073741824
              19:    7(fvec3) ConstantComposite 16 17 18
              20:             TypePointer Function 7(fvec3)
              22:      8(int) Constant 1
              23:             TypePointer Function 8(int)
    28(MyStruct):             TypeStruct 7(fvec3) 8(int)
              29:             TypePointer Function 28(MyStruct)
              31:      8(int) Constant 2
              46:             TypePointer Output 6(float)
        47(cout):     46(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
           30(s):     29(ptr) Variable Function
           33(r):     23(ptr) Variable Function
       34(param):     10(ptr) Variable Function
              32:     23(ptr) AccessChain 30(s) 22
                              Store 32 31
              35:28(MyStruct) Load 30(s)
              36:    7(fvec3) CompositeExtract 35 0
              37:     20(ptr) AccessChain 34(param) 15
                              Store 37 36
              38:      8(int) CompositeExtract 35 1
              39:     23(ptr) AccessChain 34(param) 22
                              Store 39 38
              40:      8(int) FunctionCall 13(foo(struct-MyStruct-vf3-i11;) 34(param)
              41: 9(MyStruct) Load 34(param)
              42:    7(fvec3) CompositeExtract 41 0
              43:     20(ptr) AccessChain 30(s) 15
                              Store 43 42
              44:      8(int) CompositeExtract 41 1
              45:     23(ptr) AccessChain 30(s) 22
                              Store 45 44
                              Store 33(r) 40
                              Return
                              FunctionEnd
13(foo(struct-MyStruct-vf3-i11;):      8(int) Function None 11
           12(s):     10(ptr) FunctionParameter
              14:             Label
              21:     20(ptr) AccessChain 12(s) 15
                              Store 21 19
              24:     23(ptr) AccessChain 12(s) 22
              25:      8(int) Load 24
                              ReturnValue 25
                              FunctionEnd
